src/utils/string.ts

```diff
@@ -2,6 +2,18 @@ import { Buffer } from "buffer";
 
 import { throwValidationError, validateHexString, validateString } from "./validation";
 
+/**
+ * Returns a boolean indicating whether the String contains only ASCII bytes.
+ * From https://stackoverflow.com/a/14313213/6085242.
+ *
+ * @param str - The input string.
+ * @returns - Whether the string is ASCII.
+ */
+export function isASCIIString(str: string): boolean {
+  // eslint-disable-next-line
+  return /^[\x00-\x7F]*$/.test(str);
```

Comment:
I would allow at least the values up to 165 (0xA5) because those cover all umalut character variations and the soft `n` Spanish uses. Or, if we want to be pedantic, all up to 0xFF because those are ASCII characters that we can put in a string. If we're excluding anything that should probably be the first 32 characters because those are control and non-printable characters.

Reply:
I looked up the function in Sia and it uses the same range of characters. I had a hunch that Unicode isn't compatible with all ASCII characters and that turned out to be correct -- it is only backwards compatible with the original ASCII Characters 0-127, which matches the range being used here.